### ISSUE TITLE
[Help needed] Struggle with 'cargo make'

### ISSUE THREAD

**Issue by freijon** [espanso/espanso#1595]
I'm working on a package for the Gentoo Linux distribution. If you don't know the specifics about Gentoo packaging, I'll add some information here [^1].

I'm struggling with `cargo make`...

1. First of all it seems `cargo make` tries to download & install packages on its own, they are nowhere defined in this repository. For example the crate `rust-script-0.7.0` and `envmnt-*` are needed. I worked around this by pre-downloading `rust-script-0.7.0` and `envmnt-*` and they seem to build correctly. So far so good...
2. `espanso-migrate` Cargo.toml requires the git version of `yaml-rust`. I can pre-fetch the current master of that crate and patch Cargo.toml to use the local crate, but now I get `🌐 Permission denied (os error 13)`
3. It seems in https://github.com/espanso/espanso/blob/dev/scripts/build_binary.rs#L90-L96 you are removing Cargo ENV variables, I think this messes with portage and `cargo make` will no longer work in `--offline` mode. I tried commenting these lines with no luck...

I'd really appreciate some insight on how I could easily build `espanso` in a vendored way: Pre-Download all needed ressources and build offline. Your help is really appreciated.

[^1]: Gentoo is a source based linux distro which builds all packages from source with the portage package manager. All package resources need to be fetched beforehand. For Rust packages this means all necessary crates are downloaded and verified outside of `cargo` and then `cargo build --offline` is used.

**Comment by freijon** [comment]
I managed to build the package with just the regular `cargo build` instead of `cargo make`. Do you see any negative side-effects with this approach? Will this approach also work in the future?

**Comment by federico-terzi** [comment]
Hey @freijon,

Sorry for the late response! In theory, using `cargo build` should be fine as long as you pass all the necessary parameters and env variables, as available here: https://github.com/espanso/espanso/blob/dev/scripts/build_binary.rs

Hope that helps :)

**Comment by freijon** [comment]
That's exactly what I ended up doing. Portage now executes the following command:
`cargo build -p espanso --verbose --no-default-features --features modulo native-tls (wayland?)` where the `wayland` feature is optional. It works for now and I hope it keeps working in the future ;) Unless you don't do anything fancy that really needs `cargo make` where we would miss out on, we'll just use the good old `cargo build` for now and I'm fine to close this issue.

P.S. first I also made `modulo` optional but found that `espanso` is unusable without it, so it's now enabled unconditionally.
